src/aircraft_cmd.cpp

```diff
@@ -139,10 +139,24 @@ static StationID FindNearestHangar(const Aircraft *v)
 
 		/* v->tile can't be used here, when aircraft is flying v->tile is set to 0 */
 		uint distance = DistanceSquare(vtile, st->airport.tile);
-		if (v->acache.cached_max_range_sqr != 0) {
-			/* Check if our current destination can be reached from the depot airport. */
-			const Station *cur_dest = GetTargetAirportIfValid(v);
-			if (cur_dest != NULL && DistanceSquare(st->airport.tile, cur_dest->airport.tile) > v->acache.cached_max_range_sqr) continue;
+
+		uint max_range = v->acache.cached_max_range_sqr;
+		if (max_range != 0) {
+			/* Determine destinations */
```

Comment:
I'd appreciate a comment here more thoroughly describing the situation being tested for/prevented by these checks.

Reply:
This is trying to prevent Aircraft with a max range from getting a 'destination too far' after being serviced at a nearest hangar. The nearest hangar is an order (let's call it H) that is placed between orders A and B. The code is trying to ensure that the range from H to B and H to A are still within the max range limit of the aircraft. It is based on the assumption that A to B are within the range limit.

Determining the correct last and next destinations is the complicated part, and I'm not 100% sure that part of the code is fool proof, but it's the best I could do to. `GetTargetAirportIfValid` doesn't update in the same manner as the others, and then there's the 'Service at nearest hangar' which also behaves different during the `v->current order.GetDestination()`, so sometimes the last_dest is shifted an order behind, and next_dest is shifted an order ahead.
I don't know how to make this better. From my testings, it seems to be working fine enough, but without garantees.